Compiling ncurses fails.

So whenever I do `./cook.sh ncurses build` it fails because it can't find `termios.h`, instead it tries to include `<sgtty.h>`.

I tried fixing it myself, but couldn't figure it out due to my lack of experience with autoconf, here's what I found however:

* Checking whether `termios.h`/`termio.h` exists is done somewhere in ./configure
* It created a variable called `_HAVE_TERMIOS_H` that is set to 0
* The code that is generated doesn't include termios.h, because autoconf thinks it's non-existent, instead.

Apparently I updated everything except newlib, however I have a new error now.

```
x86_64-unknown-redox-gcc -DHAVE_CONFIG_H -I. -I../include   -D_XOPEN_SOURCE=500 -DNDEBUG -O2 --param max-inline-insns-single=1200 -c ../ncurses/lib_gen.c -o ../objects/lib_gen.o
In file included from ./curses.priv.h:325:0,
                 from ../ncurses/lib_gen.c:19:
_15019.c:843:15: error: expected ')' before 'int'
../include/curses.h:1631:56: note: in definition of macro 'mouse_trafo'
 #define mouse_trafo(y,x,to_screen) wmouse_trafo(stdscr,y,x,to_screen)
                                                        ^
Makefile:962: recipe for target '../objects/lib_gen.o' failed
```

--- a/recipes/ncurses/recipe.sh
+++ b/recipes/ncurses/recipe.sh
@@ -12,6 +12,7 @@ function recipe_update {
 }
 
 function recipe_build {
+    export CPPFLAGS="-P"
     ./configure --host=${HOST} --prefix=""
     make
     skip=1

